internal/venice-common/src/main/java/com/linkedin/venice/writer/VeniceWriter.java

```diff
@@ -394,13 +407,13 @@ public void close(boolean gracefulClose) {
         // DO NOT call the {@link #PubSubProducerAdapter.close(int) version from here.}
         // For non-shared producer mode gracefulClose will flush the producer
 
-        producerAdapter.close(topicName, closeTimeOutInMs, gracefulClose);
+        getProducerAdapter().close(topicName, closeTimeOutInMs, gracefulClose);
         OPEN_VENICE_WRITER_COUNT.decrementAndGet();
       } catch (Exception e) {
         logger.warn("Swallowed an exception while trying to close the VeniceWriter for topic: {}", topicName, e);
         VENICE_WRITER_CLOSE_FAILED_COUNT.incrementAndGet();
       }
-      threadPoolExecutor.shutdown();
+      getThreadPoolExecutor().shutdown();
```

Comment:
This might need a bit improvement, `shutdown` only makes sure no new tasks will be accepted. Take a look at our conventions for close a thread pool, e.g., `VeniceParentHelixAdmin.close` or somewhere else.

Reply:
Do you mean use shutdownNow to interrupt the running threads?

Reply:
yes, shutdownnow + awaitTermination for a limited timeout.  does it make any sense?